### PR TITLE
Sync dataset name with datasource in add remote view

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- try adding a remote dataset
- enter a DS name
- submitting should work
- when checking the advanced settings tab before submitting `id.name` should be in sync with the dataset name input field of the simple settings view

### Issues:
- fixes https://scm.slack.com/archives/C5AKLAV0B/p1732788765872929

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)

### CODE REVIEW

File: frontend/javascripts/dashboard/dataset/dataset_settings_data_tab.tsx [hunk 1]
```diff
@@ -33,12 +33,16 @@ import { type APIDataLayer, type APIDataset, APIJobType } from "types/api_flow_t
...
+import type { ArbitraryObject } from "types/globals";
 
 const FormItem = Form.Item;
 
 export const syncDataSourceFields = (
   form: FormInstance,
   syncTargetTabKey: "simple" | "advanced",
+  // Syncing the dataset name is optional as this is needed for the add remote view, but not for the edit view.
+  // In the edit view, the datasource.id fields should never be changed and the backend will automatically ignore all changes to the id field.
+  syncDatasetName = false,
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/dashboard/dataset/dataset_settings_data_tab.tsx:45]: It there maybe a better place to put this comment?

Comment (philippotto) [reply frontend/javascripts/dashboard/dataset/dataset_settings_data_tab.tsx:45]: I think, it's alright.

**Review by philippotto — CHANGES_REQUESTED** [review]
code looks alright, but testing didn't work for me. in my case I did **not** switch to the advanced tab. I think in that case the synchronization is not triggered? at least, I got this error: `toast.tsx:215 The datasource config could not be stored. Error: The dataset name must not be empty.`. only after switching to the tab, the import worked.

File: frontend/javascripts/admin/dataset/dataset_add_remote_view.tsx [hunk 2]
```diff
@@ -252,13 +252,13 @@ function DatasetAddRemoteView(props: Props) {
...
     // data which is rendered by the "simple" page.
-    syncDataSourceFields(form, "simple");
+    syncDataSourceFields(form, "simple", true);
     form.validateFields();
   };
 
   async function handleStoreDataset() {
     // Sync simple with advanced and get newest datasourceJson
-    syncDataSourceFields(form, dataSourceEditMode === "simple" ? "advanced" : "simple");
+    syncDataSourceFields(form, dataSourceEditMode === "simple" ? "advanced" : "simple", true);
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/admin/dataset/dataset_add_remote_view.tsx:261]: I missed a few occurrences of `syncDataSourceFields` :see_no_evil:, this one here being the most important.

File: frontend/javascripts/dashboard/dataset/dataset_settings_data_tab.tsx [hunk 3]
```diff
@@ -47,12 +51,25 @@ export const syncDataSourceFields = (
   if (syncTargetTabKey === "advanced") {
     // Copy from simple to advanced: update json
     const dataSourceFromSimpleTab = form.getFieldValue("dataSource");
+    if (syncDatasetName && dataSourceFromSimpleTab) {
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/dashboard/dataset/dataset_settings_data_tab.tsx:54]: The `&& dataSourceFromSimpleTab` was added as `syncDataSourceFields` is called before the first exploration request is sent to the backend. In this case `dataSourceFromSimpleTab==undefined` and thus I guarded this assignment below as else the code would silently crash.

**Review by MichaelBuessemeyer — COMMENTED** [review]
Thanks for noticing :pray: . Should be better now

**Pull request merged**